src/cpp/session/modules/SessionFind.cpp

```diff
@@ -780,55 +799,81 @@ class GrepOperation : public boost::enable_shared_from_this<GrepOperation>
                const size_t matchSize = matchOff - matchOn;
                size_t replaceMatchOff = matchOff;
                Error error;
-               Replacer replacer(findResults().ignoreCase());
-               std::string newLine(pLineInfo->decodedContents);
+               Replacer replacer(encoding_, findResults().ignoreCase());
+
+               if (!encoding_.empty())
+               {
+                  eMatchOn =
+                     static_cast<size_t>(eMatchOnArray.getValueAt(static_cast<size_t>(pos)).getInt());
+                  eMatchOff =
+                     static_cast<size_t>(eMatchOffArray.getValueAt(static_cast<size_t>(pos)).getInt());
+               }
+
+               // If we found a different number of matches searching the encoded string,
+               // we shouldn't perform the replace as the expected vs actual results may differ.
+               if (!encoding_.empty() && eMatchOnArray.getSize() != matchOnArray.getSize())
+               {
+                  core::Error error(
+                     errc::findCategory(),
+                     errc::RegexError,
+                     "Could not perform replace due to encoding issue.",
```

Comment:
Could this be made more specific so that we have more to go on if it's seen in the wild? E.g.:

"Found X matches in line but Y matches in encoded line; skipping replace."

Reply:
Good idea, fixed.